frontend/src/components/mine/Permit/MinePermitTable.js

```diff
@@ -0,0 +1,467 @@
+    .sort((order1, order2) => {
+      const date1 = Date.parse(order1.due_date) || 0;
+      const date2 = Date.parse(order2.due_date) || 0;
+      return date1 === date2 ? order1.order_no - order2.order_no : date1 - date2;
+    });
+
+const getPermitteeName = (permittees) =>
+  permittees[0] ? permittees[0].party.party_name : Strings.EMPTY_FIELD;
+
+const transformRowData = (
+  permit,
+  partyRelationships,
+  major_mine_ind,
+  openEditPermitModal,
+  openAddPermitAmendmentModal,
+  openAddAmalgamatedPermitModal,
+  permitStatusOptions
+) => {
+  const latestAmendment = permit.amendments[0];
+  const firstAmendment = permit.amendments[permit.amendments.length - 1];
+
+  const permittees = getPermittees(partyRelationships, permit);
+  const permitteeName = partyRelationships.length === 0 ? "" : getPermitteeName(permittees);
+  const hasAmalgamated = permit.amendments.find((pa) => pa.permit_amendment_type_code === "ALG");
+
+  return {
+    key: permit.permit_guid,
+    lastAmended:
+      (latestAmendment && latestAmendment.issue_date && formatDate(latestAmendment.issue_date)) ||
+      Strings.EMPTY_FIELD,
+    permitNo: permit.permit_no || Strings.EMPTY_FIELD,
+    firstIssued:
+      (firstAmendment && firstAmendment.issue_date && formatDate(firstAmendment.issue_date)) ||
+      Strings.EMPTY_FIELD,
+    permittee: permitteeName,
+    authorizationEndDate:
+      latestAmendment && latestAmendment.authorization_end_date
+        ? formatDate(latestAmendment.authorization_end_date)
+        : Strings.EMPTY_FIELD,
+    amendments: permit.amendments,
+    status: permit.permit_status_code,
+    addEditButton: {
+      guid: permit.permit_guid,
+      major_mine_ind,
+      hasAmalgamated,
+      permit_no: permit.permit_no,
+    },
+    openEditPermitModal,
+    openAddPermitAmendmentModal,
+    openAddAmalgamatedPermitModal,
+    permitStatusOptions,
+  };
+};
+
+const transformChildRowData = (
+  amendment,
+  record,
+  amendmentNumber,
+  major_mine_ind,
+  openEditAmendmentModal
+) => ({
+  amendmentNumber,
+  amendmentType: amendment.permit_amendment_type_code,
+  receivedDate:
+    (amendment.received_date && formatDate(amendment.received_date)) || Strings.EMPTY_FIELD,
+  issueDate: (amendment.issue_date && formatDate(amendment.issue_date)) || Strings.EMPTY_FIELD,
+  authorizationEndDate:
+    (amendment.authorization_end_date && formatDate(amendment.authorization_end_date)) ||
+    Strings.EMPTY_FIELD,
+  description: amendment.description || Strings.EMPTY_FIELD,
+  amendmentEdit: {
+    guid: amendment.permit_amendment_guid,
+    permit_guid: amendment.permit_guid,
+    major_mine_ind,
+    description: amendment.description,
+  },
+  openEditAmendmentModal,
+});
+
+export const MinePermitTable = (props) => {
+  // Edit permit Amendment
+  const handleEditPermitAmendment = (data) => {
+    props.updatePermitAmendment(data.permit_amendment_guid, data).then(() => {
+      props.closeModal();
```

Comment:
Not a big deal, but the outer function is unnecessary here:
```javascript
.then(props.closeModal);
```

Reply:
True. Modified.